includes/Modules/Analytics/Advanced_Tracking/Measurement_Code_Injector.php

```diff
@@ -0,0 +1,122 @@
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Analytics\Advanced_Tracking;
+
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\Measurement_Event_Pipe;
+
+/**
+ * Class for injecting Javascript based on the current active plugins.
+ *
+ * @since n.e.x.t.
+ * @access private
+ * @ignore
+ */
+final class Measurement_Code_Injector {
+
+	/**
+	 * Holds a list of event configurations to be injected
+	 *
+	 * @since n.e.x.t.
+	 * @var array
+	 */
+	private $event_configurations;
+
+	/**
+	 * The javascript code that is injected for event tracking.
+	 *
+	 * @since n.e.x.t.
+	 * @var string
+	 */
+	protected $inject_script;
+
+	/**
+	 * Injector constructor.
+	 *
+	 * @since n.e.x.t.
+	 *
+	 * @param array $event_configurations list of measurement events to track.
+	 */
+	public function __construct( $event_configurations ) {
+		$this->event_configurations = Measurement_Event_Pipe::encode_measurement_event_list( $event_configurations );
+		$this->inject_script        = <<<INJECT_SCRIPT
+( function() {
+    var eventConfigurations = {$this->event_configurations};
+	var config;
+	for ( config of eventConfigurations ) {
+		const thisConfig = config;
+		document.addEventListener( config.on, function( e ) {
+			var el = e.target;
+			var matcher = el.matches || el.webkitMatchesSelector || el.mozMatchesSelector || el.msMatchesSelector || el.oMatchesSelector;
+			if ( matcher && ( matcher.call( el, thisConfig.selector ) || matcher.call( el, thisConfig.selector.concat( ' *' ) ) ) ) {
+				alert( 'Got an event called: '.concat( thisConfig.action ) );
+
+				var params = {};
+				if ( "metadata" in thisConfig && null !== thisConfig.metadata ) {
+					params = thisConfig.metadata( params, el );
+				}
+				params['event_category'] = thisConfig.category;
+				console.log(params);
+
+				gtag( 'event', thisConfig.action, params );
+			}
+		}, true );
+	}
+	jQuery(document).ready(function(){
+		if( null !== woocommerceOrderData.transaction_id ) {
+			alert( 'Got an event called: purchase!!!' );
+
+			var params = {};
+			var transaction_id = woocommerceOrderData.transaction_id;
+			if ( '' === transaction_id ) {
+				transaction_id = 'not_paid';
+			}
+			params['transaction_id'] = transaction_id;
+			params['value'] = woocommerceOrderData.subtotal;
+			params['currency'] = woocommerceOrderData.currency;
+			params['tax'] = parseFloat( woocommerceOrderData.subtotal_tax ) + parseFloat( woocommerceOrderData.shipping_tax );
+			params['shipping'] = parseFloat( woocommerceOrderData.shipping );
+			var items = [];
+			for ( itemName in woocommerceOrderData.items ) {
+				items.push( woocommerceOrderData.items[ itemName ] );
+			}
+			params['items'] = items;
+			params['event_category'] = 'ecommerce';
+
+		    console.log(params);
+
+			gtag( 'event', 'purchase', params );
+		}
+	});
```

Comment:
The issue has been resolved. Because this functionality is out of the scope of this PR, we have moved the issue out of the current PR and left a TODO comment.